redis/src/commands/mod.rs

```diff
@@ -2166,6 +2166,73 @@ impl PubSubCommands for Connection {
     }
 }
 
+/// Options for the [SCAN](https://redis.io/commands/scan) command
+///
+/// # Example
+///
+/// ```rust,no_run
+/// use redis::{Commands, RedisResult, ScanOptions};
+/// fn force_fetching_every_matching_key(
+///     con: &mut redis::Connection,
+///     pattern: Option<&str>,
+///     cursor: Option<u64>,
+///     count: Option<usize>,
+/// ) -> RedisResult<Vec<usize>> {
+///     let opts = ScanOptions::default()
+///         .cursor(cursor.unwrap_or_default())
+///         .pattern(pattern)
+///         .count(count);
+///     con.scan_options(opts)
+/// }
+/// ```
+#[derive(Default)]
+pub struct ScanOptions {
+    cursor: u64,
+    pattern: Option<String>,
+    count: Option<usize>,
+}
+
+impl ScanOptions {
+    /// Limit the results to the first N matching items.
+    pub fn count(mut self, n: usize) -> Self {
```

Comment:
please use `set_X` instead of `X` for the options function names. I know that we're not consistent on this, I try to improve that in the library.

Reply:
how about `with_*`? its a general convention in rust libs for builder pattern: `Options::default().with_field(a)`

Reply:
sure